dev/com.ibm.ws.microprofile.health/src/com/ibm/ws/microprofile/health/internal/AppTrackerImpl.java

```diff
@@ -238,6 +253,63 @@ public boolean isStarted(String appName) {
         }
     }
 
+    /**
+     * Returns true if the application with the specified name is installed, otherwise false.
+     *
+     * @return true if the application with the specified name is installed, otherwise false.
+     */
+    @Override
+    public boolean isInstalled(String appName) {
+        lock.readLock().lock();
+        try {
+            if (appStateMap.get(appName) == ApplicationState.INSTALLED) {
+                String state = getApplicationMBean(appName);
+                if (state.isEmpty()) {
+                    appStateMap.replace(appName, null);
+                } else {
+                    return true;
+                }
+            }
+        } finally {
+            lock.readLock().unlock();
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the application with the specified name is uninstalled, otherwise false.
+     *
+     * @return true if the application with the specified name is uninstalled, otherwise false.
+     */
+    @Override
+    public boolean isUninstalled(String appName) {
+        lock.readLock().lock();
+        try {
+            if (appStateMap.get(appName) == null) {
+                return true;
+            }
+        } finally {
+            lock.readLock().unlock();
+        }
+        return false;
+    }
+
+    /**
+     * Returns the MBeanInfo of appName if the ApplicationMBean exists, otherwise null.
+     *
+     * @return the MBeanInfo of appName if the ApplicationMBean exists, otherwise null.
+     */
+    private String getApplicationMBean(String appName) {
+        MBeanInfo bean = null;
+        String state = "";
+        try {
+            ObjectName objectName = new ObjectName("WebSphere:service=com.ibm.websphere.application.ApplicationMBean,name=" + appName);
+            bean = mbeanServer.getMBeanInfo(objectName);
+            state = (String) mbeanServer.getAttribute(objectName, "State");
+        } catch (Exception e) {}
```

Comment:
Add a Tr.debug() in the catch block and print out the Exception message, so we can debug it, if something goes wrong.

Reply:
Added a Tr.debug message.